File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int answer = 0; 
        int bestPrice = prices[0]; 
        for(int i = 1; i < prices.size(); i++){
            int curr = prices[i]; 
            if(curr - bestPrice > 0){
                answer += (curr - bestPrice); 
                bestPrice = INT_MAX; 
            }
            bestPrice = min(bestPrice, curr); 
        }
        return answer; 
    }
};

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int[] dp = new int[prices.length]; 
        dp[0] = 0; 
        for(int i = 1; i < prices.length; i++){
            dp[i] = dp[i-1] + Math.max(0,prices[i] - prices[i-1]); 
        }
        return dp[prices.length-1]; 
    }
}